data/src/main/java/com/ericsson/bss/cassandra/ecchronos/data/sync/EccNodeMetadata.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2024 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecchronos.data.sync;
+
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.BoundStatement;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.querybuilder.QueryBuilder;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.selectFrom;
+
+public final class EccNodeMetadata
+{
+
+    private static final Logger LOG = LoggerFactory.getLogger(EccNodeMetadata.class);
+    private static final String COLUMN_ECCHRONOS_ID = "ecchronos_id";
+    private static final String COLUMN_DC_NAME = "datacenter_name";
+    private static final String COLUMN_NODE_ID = "node_id";
+    private static final String KEYSPACE_NAME = "ecchronos";
+    private static final String NODE_METADATA_TABLE_NAME = "nodes_metadata";
+
+    private final PreparedStatement mySelectStatement;
+    private final PreparedStatement myCreateStatement;
+    private final PreparedStatement myTableUpdateStatement;
+    private final CqlSession mySession;
+    private final Node myNode;
+
+    public EccNodeMetadata(final List<Node> nodes, final CqlSession session)
+    {
+        mySession = session;
+        myNode = nodes.stream().findAny().get();
```

Comment:
What happens if the topology changes and you get a node that was not on the first configuration for a instance that was working?

let's say that in the configuration we are using datacenterAware, so we have a datacenter list of [d1, dc2] but now the topology changed and we added dc3 on the list, but it was not on the previously topology for this instance, so if this line picks up a node from dc3 it will generate a new ecchronos id for a instance that already has an ID